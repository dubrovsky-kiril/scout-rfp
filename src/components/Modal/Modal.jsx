import React, { memo } from "react";
import PropTypes from "prop-types";
import Button from "#components/Button/Button";
import styles from "./Modal.scss";

const Modal = ({ title, msg, onClick, buttonTxt }) => {
  return (
    <div className={styles.overlay}>
      <div className={styles.modal}>
        <h2>{title}</h2>
        <p>{msg}</p>
        <Button onClick={onClick}>{buttonTxt}</Button>
      </div>
    </div>
  );
};

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  msg: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  buttonTxt: PropTypes.string.isRequired
};

export default memo(Modal);
