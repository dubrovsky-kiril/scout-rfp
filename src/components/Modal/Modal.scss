.overlay {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  padding: 1rem;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 9999;
  opacity: 1;
  overflow-x: hidden;
  overflow-y: auto;
  animation: show 0.5s ease;
  .modal {
    text-align: center;
    h2 {
      margin-top: 0;
    }
    p {
      margin-top: 0;
      margin-bottom: 30px;
    }
    border-radius: 8px;
    z-index: 1;
    background-color: #fff;
    color: #222222;
    position: fixed;
    margin-left: auto;
    margin-right: auto;
    display: block;
    top: calc(50% - 100px);
    max-width: 400px;
    width: 100%;
    padding: 30px 20px;
  }
}
