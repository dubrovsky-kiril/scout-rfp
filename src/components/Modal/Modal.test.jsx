import React from "react";
import { create } from "react-test-renderer";
import Modal from "./Modal";

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(
      <Modal title="Title" msg="Message" buttonTxt="Click" onClick={() => {}} />
    );

    expect(button.toJSON()).toMatchSnapshot();
  });
  test("Is showing exptected title", () => {});
  test("Is showing exptected message", () => {});
  test("Is showing exptected button text", () => {});
  test("Is passing onClick to Button", () => {});
});
