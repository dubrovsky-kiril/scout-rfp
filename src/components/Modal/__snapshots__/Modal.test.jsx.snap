// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Matches the snapshot 1`] = `
<div
  className="overlay"
>
  <div
    className="modal"
  >
    <h2>
      Title
    </h2>
    <p>
      Message
    </p>
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Click
    </button>
  </div>
</div>
`;
